### ISSUE TITLE
E_PROXY_COOKIE_AUTHENTICATOIN_ACCESS_DENIED with RDS gateway

### ISSUE THREAD

**Issue by joeydumont** [FreeRDP/FreeRDP#5581]
Trying to connect using an RDS-generated RDP file triggers a `E_PROXY_COOKIE_AUTHENTICATION_ACCESS_DENIED` error. 

I recompiled FreeRDP with `DWITH_GSSAPI=OFF` as per #4311, to no avail (see below).

**To Reproduce**
  1. Download the RDP file from your Windows Server 2012 R2 RDS server.
  2. Try to connect with `xfreerdp rds-file.rdp /u:<username> /d:<domain> /sec:nla /log-level:TRACE`.
  3. See the following error:
```console
[15:03:42:036] [5875:5877] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpdr
[15:03:42:036] [5875:5877] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpsnd
[15:03:42:036] [5875:5877] [DEBUG][com.freerdp.channels.cliprdr.client] - VirtualChannelEntryEx
[15:03:42:036] [5875:5877] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[15:03:42:036] [5875:5877] [DEBUG][com.freerdp.channels.rail.client] - VirtualChannelEntryEx
[15:03:42:036] [5875:5877] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rail
[15:03:42:036] [5875:5877] [DEBUG][com.freerdp.channels.drdynvc.client] - VirtualChannelEntryEx
[15:03:42:036] [5875:5877] [INFO][com.freerdp.client.common.cmdline] - loading channelEx drdynvc
[15:03:42:066] [5875:5877] [DEBUG][com.freerdp.client.x11] - Get global pointer mapping (no XInput)
[15:03:42:069] [5875:5877] [DEBUG][com.freerdp.core.nego] - Enabling security layer negotiation: TRUE
[15:03:42:069] [5875:5877] [DEBUG][com.freerdp.core.nego] - Enabling restricted admin mode: FALSE
[15:03:42:069] [5875:5877] [DEBUG][com.freerdp.core.nego] - Enabling RDP security: FALSE
[15:03:42:069] [5875:5877] [DEBUG][com.freerdp.core.nego] - Enabling TLS security: FALSE
[15:03:42:069] [5875:5877] [DEBUG][com.freerdp.core.nego] - Enabling NLA security: TRUE
[15:03:42:069] [5875:5877] [DEBUG][com.freerdp.core.nego] - Enabling NLA extended security: FALSE
[15:03:42:069] [5875:5877] [DEBUG][com.freerdp.core.nego] - state: NEGO_STATE_NLA
[15:03:42:069] [5875:5877] [DEBUG][com.freerdp.core.nego] - Attempting NLA security
[15:03:42:074] [5875:5877] [DEBUG][com.freerdp.core] - connecting to peer <gw.ip.ad.dr>
[15:03:42:138] [5875:5877] [WARN][com.freerdp.crypto] - Certificate verification failure 'unable to get local issuer certificate (20)' at stack position 1
[15:03:42:139] [5875:5877] [WARN][com.freerdp.crypto] - C = US, O = "Entrust, Inc.", OU = See www.entrust.net/legal-terms, OU = "(c) 2012 Entrust, Inc. - for authorized use only", CN = Entrust Certification Authority - L1K
[15:03:42:157] [5875:5877] [DEBUG][com.freerdp.core.gateway.rdg] - RDG_OUT_DATA authorization result: 200
[15:03:42:157] [5875:5877] [DEBUG][com.freerdp.core] - connecting to peer <gw.ip.ad.dr>
[15:03:43:232] [5875:5877] [DEBUG][com.freerdp.core.gateway.rdg] - RDG_IN_DATA authorization result: 200
[15:03:43:249] [5875:5877] [DEBUG][com.freerdp.core.gateway.rdg] - Handshake response received
[15:03:43:249] [5875:5877] [DEBUG][com.freerdp.core.gateway.rdg] - errorCode=RPC_S_OK [0x00000000], verMajor=1, verMinor=0, serverVersion=0, extendedAuth=HTTP_EXTENDED_AUTH_SC|HTTP_EXTENDED_AUTH_PAA|HTTP_EXTENDED_AUTH_SSPI_NTLM [0007]
[15:03:43:347] [5875:5877] [DEBUG][com.freerdp.core.gateway.rdg] - Tunnel response received
[15:03:43:347] [5875:5877] [DEBUG][com.freerdp.core.gateway.rdg] - serverVersion=0, errorCode=E_PROXY_COOKIE_AUTHENTICATION_ACCESS_DENIED [0x800759F8], fieldsPresent= [0000]
[15:03:43:347] [5875:5877] [ERROR][com.freerdp.core.gateway.rdg] - Tunnel creation error E_PROXY_COOKIE_AUTHENTICATION_ACCESS_DENIED [0x800759F8]
[15:03:43:347] [5875:5877] [ERROR][com.freerdp.core.nego] - Protocol Security Negotiation Failure
[15:03:43:347] [5875:5877] [ERROR][com.freerdp.core] - freerdp_set_last_error ERRCONNECT_SECURITY_NEGO_CONNECT_FAILED [0x0002000C]
[15:03:43:347] [5875:5877] [ERROR][com.freerdp.core.connection] - Error: protocol security negotiation or connection failure
```

**Expected behavior**
Expected to be connected to a remote desktop.

**Application details**
* Version of FreeRDP: `2.0.0-dev5 (79d19b5f7)`
* Command line used: `xfreerdp rds-file.rdp /u:<username> /d:<domain> /sec:nla /log-level:TRACE`
* output of `/buildconfig`: 
```xfreerdp /buildconfig
This is FreeRDP version 2.0.0-dev5 (79d19b5f7)
Build configuration: BUILD_TESTING=OFF BUILTIN_CHANNELS=ON HAVE_AIO_H=1 HAVE_EXECINFO_H=1 HAVE_FCNTL_H=1 HAVE_INTTYPES_H=1 HAVE_MATH_C99_LONG_DOUBLE=1 HAVE_POLL_H=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK=ON HAVE_PTHREAD_MUTEX_TIMEDLOCK_LIB=1 HAVE_PTHREAD_MUTEX_TIMEDLOCK_SYMBOL= HAVE_SYSLOG_H=1 HAVE_SYS_EVENTFD_H=1 HAVE_SYS_FILIO_H= HAVE_SYS_MODEM_H= HAVE_SYS_SELECT_H=1 HAVE_SYS_SOCKIO_H= HAVE_SYS_STRTIO_H= HAVE_SYS_TIMERFD_H=1 HAVE_TM_GMTOFF=1 HAVE_UNISTD_H=1 WITH_ALSA=OFF WITH_CAIRO=OFF WITH_CCACHE=ON WITH_CHANNELS=ON WITH_CLANG_FORMAT=ON WITH_CLIENT=ON WITH_CLIENT_AVAILABLE=1 WITH_CLIENT_CHANNELS=ON WITH_CLIENT_CHANNELS_AVAILABLE=1 WITH_CLIENT_COMMON=ON WITH_CLIENT_INTERFACE=OFF WITH_CUPS=OFF WITH_DEBUG_ALL=OFF WITH_DEBUG_CAPABILITIES=OFF WITH_DEBUG_CERTIFICATE=OFF WITH_DEBUG_CHANNELS=OFF WITH_DEBUG_CLIPRDR=OFF WITH_DEBUG_DVC=OFF WITH_DEBUG_KBD=OFF WITH_DEBUG_LICENSE=OFF WITH_DEBUG_MUTEX=OFF WITH_DEBUG_NEGO=OFF WITH_DEBUG_NLA=OFF WITH_DEBUG_NTLM=OFF WITH_DEBUG_RAIL=OFF WITH_DEBUG_RDP=OFF WITH_DEBUG_RDPDR=OFF WITH_DEBUG_RDPEI=OFF WITH_DEBUG_RDPGFX=OFF WITH_DEBUG_REDIR=OFF WITH_DEBUG_RFX=OFF WITH_DEBUG_RINGBUFFER=OFF WITH_DEBUG_SCARD=OFF WITH_DEBUG_SND=OFF WITH_DEBUG_SVC=OFF WITH_DEBUG_SYMBOLS=OFF WITH_DEBUG_THREADS=OFF WITH_DEBUG_TIMEZONE=OFF WITH_DEBUG_TRANSPORT=OFF WITH_DEBUG_TSG=OFF WITH_DEBUG_TSMF=OFF WITH_DEBUG_WND=OFF WITH_DEBUG_X11=OFF WITH_DEBUG_X11_CLIPRDR=OFF WITH_DEBUG_X11_LOCAL_MOVESIZE=OFF WITH_DEBUG_XV=OFF WITH_DSP_EXPERIMENTAL=OFF WITH_DSP_FFMPEG=Off WITH_EVENTFD_READ_WRITE=1 WITH_FAAC=OFF WITH_FAAD2=OFF WITH_FFMPEG=OFF WITH_GFX_H264=OFF WITH_GPROF=OFF WITH_GSM=OFF WITH_GSSAPI=OFF WITH_GSTREAMER_0_10=OFF WITH_GSTREAMER_1_0=OFF WITH_ICU=OFF WITH_IPP=OFF WITH_JPEG=OFF WITH_LAME=OFF WITH_LIBRARY_VERSIONING=ON WITH_LIBSYSTEMD=OFF WITH_MACAUDIO=OFF WITH_MACAUDIO=OFF WITH_MACAUDIO_AVAILABLE=0 WITH_MANPAGES=ON WITH_MBEDTLS=OFF WITH_OPENH264=OFF WITH_OPENSLES=OFF WITH_OPENSSL=ON WITH_OSS=ON WITH_PCSC=OFF WITH_PROFILER=OFF WITH_PULSE=ON WITH_SAMPLE=OFF WITH_SANITIZE_ADDRESS=OFF WITH_SANITIZE_ADDRESS_AVAILABLE=1 WITH_SANITIZE_MEMORY=OFF WITH_SANITIZE_MEMORY_AVAILABLE=1 WITH_SANITIZE_THREAD=OFF WITH_SANITIZE_THREAD_AVAILABLE=1 WITH_SERVER=OFF WITH_SERVER_INTERFACE=ON WITH_SMARTCARD_INSPECT=OFF WITH_SOXR=OFF WITH_SSE2=ON WITH_SWSCALE=OFF WITH_THIRD_PARTY=OFF WITH_VALGRIND_MEMCHECK=OFF WITH_VALGRIND_MEMCHECK_AVAILABLE=1 WITH_WAYLAND=OFF WITH_WINPR_TOOLS=ON WITH_X11=ON WITH_X264=OFF WITH_XCURSOR=ON WITH_XEXT=ON WITH_XFIXES=ON WITH_XI=OFF WITH_XINERAMA=OFF WITH_XKBFILE=OFF WITH_XRANDR=OFF WITH_XRENDER=ON WITH_XSHM=ON WITH_XV=OFF WITH_ZLIB=ON
Build type:          Release
CFLAGS:               -fPIC -Wall -Wno-unused-result -Wno-unused-but-set-variable -Wno-deprecated-declarations -fvisibility=hidden -Wimplicit-function-declaration -Wredundant-decls -DWINPR_DLL
Compiler:            GNU, 4.8.5
Target architecture: x64
```
* OS version connecting to: Windows Server 2012 R2

**Desktop (please complete the following information):**
 - OS: RHEL 7

**Comment by akallabeth** [comment]
Certificate can not be validated.
Looks like you have an abort condition in the rdp file?
Is the cert authority installed locally?

**Comment by joeydumont** [comment]
Doing `/cert-ignore` does not solve the issue.

I just noticed that the test that failed actually launched a specific application (wordpad), so maybe that's the problem. I tested connecting to a server that didn't launch an app and it worked. Below, Griffin is the name of the server, and wordpad is an app.

```diff
diff ~/Downloads/cpub-Griffin-CmsRdsh.rdp Downloads/cpub-wordpad-CmsRdsh.rdp 
22,23c22,23
< alternate shell:s:||Griffin
< remoteapplicationprogram:s:||Griffin
---
> alternate shell:s:||wordpad
> remoteapplicationprogram:s:||wordpad
25c25
< remoteapplicationname:s:Griffin
---
> remoteapplicationname:s:WordPad
```

However, it is very spotty. Sometimes the connection succeeds (I can see the heartbeat in the console), but I don't see the remote desktop window on my desktop. It is very slow compared to RDP'ing directly to the gateway and then to the box. But this is a different issue, so I guess this issue can be closed.
